app/models/user/permissions.rb

```diff
@@ -39,27 +40,53 @@ def admin_sections=(sections)
     keep_permissions = existing_permissions.values_at(*keep)
     new_permissions = create.map { |section| member_permissions.build(admin_section: section) }
 
-    self.member_permissions = keep_permissions + new_permissions
+    all_permissions = keep_permissions + new_permissions
+    # keep service permissions as they were
+    all_permissions << services_member_permission if services_member_permission
+
+    self.member_permissions = all_permissions
   ensure
     @_admin_sections = nil
   end
 
-  def member_permission_service_ids=(service_ids)
-    services_section = [:services]
+  # returns all permissions (:portal, :finance, :settings, :partners, :monitoring, :plans) for admins
+  # and the allowed ones for member users
+  def member_permission_ids
+    admin? ? AdminSection.permissions : admin_sections - [:services]
+  end
 
+  def member_permission_ids=(roles)
+    self.admin_sections = Array(roles).reject(&:blank?)
+  end
+
+  def member_permission_service_ids=(service_ids)
     if service_ids.present?
-      self.admin_sections = admin_sections | services_section
-      services_member_permission.service_ids = service_ids
-    else
-      self.admin_sections = admin_sections - services_section
+      service_ids = Array(service_ids).reject(&:blank?).map(&:to_i)
+      if services_member_permission
+        services_member_permission.service_ids = service_ids & existing_service_ids
```

Comment:
Previously, when setting service IDs, *any* ID could be stored, even if such ID doesn't exist in the account. The `member_permission_service_ids` getter though did filter the service list to include only the existing ones.
So, you could set `service_ids` to [1,2,30,40,50,60], and it would be stored in the DB, even though you only have 2 services: 1 and 2.
This fixes it.

There is still one situation when the `service_id` may include non-existing services, and it's when a service, to which permission was previously granted, is deleted. This field is not updated automatically to remove the service ID. And we actually need this for the notifications (we need to only send notifications on service deletion to the users who previously were allowed to see this service). But on the next update of this field the dead services will be filtered out.